### PR TITLE
POC ToC

### PR DESCRIPTION
**Pull request by mruwnik**
based on [this doc](https://docs.google.com/document/d/1eyHgAxiSji4FKh9tYYGvmXb24HnwtKC11IlJEwfwt_U/edit)

image

An interactive version deployed to: https://stampy-ui.ai688.workers.dev/toc

I couldn't be bothered to keep the navigation bar floating on the left, so for now it's just a normal element. Should be ok, seeing as this is just to check if it makes sense?

### CODE REVIEW

File: app/routes/toc.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const fetcher = async () => {
+      const response = await fetch('/toc.json')
+      setToc(await response.json())
+    }
+    fetcher()
+  }, [])
+
+  return (
+    <>
+      <style>{`
```
Comment (mruwnik) [inline app/routes/toc.tsx:101]: this is fun :D

File: app/routes/toc.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        body { max-width: 1400px; }
+            `}</style>
```
Comment (Aprillion) [inline app/routes/toc.tsx:103]:
```suggestion
      <style>
        body { max-width: 1400px; }
      </style>
```

File: app/routes/toc.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  return (
+    <article className="toc" style={{padding: '10px'}}>
+      {toc.map((main: Item, i: number) => (
+        <Section key={i} item={main} />
+      ))}
+    </article>
+  )
+}
+
+export default function App() {
```
Comment (Aprillion) [inline app/routes/toc.tsx:88]:
```suggestion
export default function Toc() {
```

File: public/toc.json
Comment (Aprillion) [inline public/toc.json:1]: were any plans discussed where to host the JSON file long-term? or would it change rarely enough to keep it in git?

**Review by Aprillion — APPROVED** [review]

File: public/toc.json
Comment (mruwnik) [reply public/toc.json:1]: so far it was mainly to just see if this even makes sense - my assumption is that if it does, then it's probably best for it to be a Coda column (e.g. `parent`) and have the UI backend construct the map of questions. Or something like that

**Comment by mruwnik** [comment]
changed to:

image

**Review by ccstan99 — COMMENTED** [review]
This layout looks really promising! A few suggestions that should hopefully be easy fixes:
- Text on navigation menu should all be same size. There's no need for headers to be larger since the organization is easily discernible now
- I would start all menu items collapsed by default for cleaner, less overwhelming visual display
- Instead of starting with a blank page, pick an answer to display in the main panel by default, maybe "What is AI safety?" Eventually, this could be some kind of welcome / overview text.

File: app/routes/toc.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+  return (
+    <>
+      <style>{`
+        body { max-width: 1400px; }
+            `}</style>
+      <Header />
+      <main className="toc-container" onClick={handleSpecialLinks}>
+        <ToC toc={toc} selectItem={selectItem} />
+        <div style={{flex: '5 3 auto'}}>
```
Comment (Aprillion) [inline app/routes/toc.tsx:153]: I would use either `.toc-container {...}` in global styles or use inline styles, but let's not combine both approaches in the same compoment, please

**Review by Aprillion — APPROVED** [review]

File: app/routes/toc.tsx (hunk 4 above)
Comment (mruwnik) [reply app/routes/toc.tsx:153]: agreed, but can't be bothered to now :P

**Pull request merged**